### PR TITLE
Stp 831 refactor model in typescript part3

### PR DESCRIPTION
**Pull request by bertrandmc**
# Description

Ticket STP-831 - Refactor existing Javascript code into Typescript (Part 3)

- Large files with multiple classes were broken down (one class per file).
- Adjust folder structure to contextualize codebase. 
- Typescriptify.
- More unit tests.
- Replace default exports with named to keep API consistent on userland.
- Rename classes to improve developer understanding of their functionality when used outside the model workspace.
- Types in types.ts to follow TS best practice.
- API is exposed via index.ts to facilitate context understanding and control API exposure. 
- Improve Error messages in AbstractCondition class.
- Refactor condition-values registration to allow better file separation.
- Rename `/condition/valueFrom` to reduce ambiguity with `/values/valuesFrom`.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] I have added tests that prove my changes are effective.
- [X] New and existing unit tests pass locally with my changes

# Checklist:

- [X] My changes do not introduce any new linting errors
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [N/A] I have made corresponding changes to the documentation and versioning
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts

### CODE REVIEW

File: model/src/components/types.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    rows?: string; // when MultilineTextField
+  };
+  schema?: {
+    length?: string;
+    max?: string;
+    min?: string;
+    regex?: string;
+  };
+  content?: string; // e.g: paragraphs have content
+  values?: ComponentValues;
```
Comment (jenbutongit) [inline model/src/components/types.ts:60]: I wonder if we should split out this type for fields/lists/content? (wont block, we can add another issue for this)

File: model/src/conditions/condition-value-abstract.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Registration } from "./condition-value-registration";
+
+export class ConditionValueAbstract {
```
Comment (jenbutongit) [inline model/src/conditions/condition-value-abstract.ts:3]: #180

File: model/src/form/form-configuration.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// TODO: This doesn't seems to be used anywhere, dead legacy code to delete?
+
 export class FormConfiguration {
```
Comment (jenbutongit) [inline model/src/form/form-configuration.ts:3]: 🤔

Comment (bertrandmc) [reply model/src/form/form-configuration.ts]: Forgot that note, removed. Thanks.

**Review by jenbutongit — APPROVED** [review]

**Comment by jenbutongit** [comment]
nice!!! 🚀

File: model/src/components/types.ts (hunk 1 above)
Comment (bertrandmc) [reply model/src/components/types.ts:60]: Yes we should, let's do another ticket and also some documentation in the model's README.

**Pull request merged**